Design rules/Diversen.md

```diff
@@ -0,0 +1,190 @@
+
+bijvoorbeeld datum "0000-00-00" om aan te geven dat een datum onbekend is
+bijvoorbeeld landcode "0000" om aan te geven dat het land onbekend is
+
+Ratio: Als er informatie beschikbaar is moet die als zondanig onderkend en gemodelleerd worden.
+
+Datum opname : 29-04-2021
+Datum wijziging : 29-04-2021
+
+### DR4.3 De description van een property moet semantisch overeenkomen met de betekenis van het gegeven in een gegevenswoordenboek (infromatiemodel)
+
+We nemen bij een property een description op die semantisch overeenkomt met de beschrijving in het gegevenswoordenboek. Deze kan ingekort, vereenvoudigd, of uitgebreid zijn, maar mag de betekenis van het gegeven niet laten afwijken van de betekenis van het corresponderende gegeven in het gegevenswoordenboek.
+
+De description kan worden weggelaten wanneer evident is dat de gebruikers van de API uit de propertynaam weten wat bedoeld wordt (bijvoorbeeld huisnummer).
+
+_**Ratio:**_ Om de description leesbaar te houden voor developers kan ervoor gekozen worden deze in te korten of
+binnen de context te vereenvoudigen.
+
+Datum opname : 29-04-2021
+Datum wijziging : 29-04-2021
+
+## DR4.4 Als meerdere componenten dezelfde properties bevatten (generalisatie) gebruik dan de 'allOf' om deze samenhang vorm te geven.
+
+Wanneer er meerdere componenten zijn met meerdere dezelfde properties, moet er hergebruik worden gemaakt via een 'allOf' constructie. Dit sluit aan op object oriëntatie in de verschillende programmeeromgevingen.
+
+Bijvoorbeeld een woonadres en een postadres zijn identiek, behalve dat postadres ook een postbusnummer kent. Dan is postadres een extensie op woonadres.
+Bijvoorbeeld een natuurlijk persoon en een niet-natuurlijk persoon hebben beide een naam en een adres, maar beide hebben ook eigen gegevens (natuurlijk persoon heeft geboortedatum, niet-natuurlijk persoon heeft eigenaar), dan zijn beide een extensie op een bovenliggende component "Persoon".
+Bijvoorbeeld bij een zakelijkGerechtigde worden alleen minimale identificerende gegevens van een persoon opgenomen (alleen naam en identificatie), maar bij de persoon (resource) worden meer eigenschappen van de persoon opgenomen (zoals adres). Dan gebruikt zakelijkGerechtigde het component "persoonBeperkt" en is de uitgebreide persoon een extensie hierop.
+
+Voorbeeld:
+
+```
+     Adres:
+      type: "object"
+      properties:
+        straat:
+          type: "string"
+          example: "Nassaulaan"
+        huisnummer:
+          type: "integer"
+          example: 12
+        postcode:
+          type: "string"
+          example: "2514JS"
+        woonplaats:
+          type: "string"
+          example: "Den Haag"
+
+      PostAdres:
+       allOf:
+         - $ref: "#/components/schemas/Adres"
+         - properties:
+             postbusnummer:
+               type: "integer"
+               example: 300
+```
+
+## DR4.5 Plaats bij het gebruik van 'allOf' het hergebruikte component als eerste.
```

Comment:
Voeg 1 '#' toe anders staat deze paragraaf op hetzelfde niveau als de paragraaf in regel 6.